rest/data/patch.go

```diff
@@ -197,35 +197,57 @@ func AbortPatchesFromPullRequest(ctx context.Context, event *github.PullRequestE
 	return nil
 }
 
-// GetPatchRawPatches fetches the raw patches for a patch
-func GetPatchRawPatches(patchID string) (map[string]string, error) {
+// RawPatch contains a patch diff along with its module diffs.
+type RawPatch struct {
+	Patch      RawModule   `json:"patch"`
+	RawModules []RawModule `json:"raw_modules"`
+}
+
+// RawModule contains a module diff.
+type RawModule struct {
+	Name    string `json:"name"`
+	Diff    string `json:"diff"`
+	Githash string `json:"githash"`
+}
+
+// GetRawPatches fetches the raw patches for a patch.
+func GetRawPatches(patchID string) (*RawPatch, error) {
+	var rawPatch RawPatch
 	patchDoc, err := patch.FindOneId(patchID)
 	if err != nil {
-		return nil, gimlet.ErrorResponse{
+		return &rawPatch, gimlet.ErrorResponse{
```

Comment:
Same comment as in the other place - return nil in the error cases is more idiomatic, unless somehow a pointer to an empty struct has some special meaning that's distinct from nil.